### PR TITLE
Use psutil.cpu_count(logical=False) in auto_detect_cpus()

### LINKED ISSUE #553: When using "-n auto", count the number of physical CPU cores instead of logical ones
I have a laptop with 6 physical CPU cores and hyper-threading. The code from [auto_detect_cpus](https://github.com/pytest-dev/pytest-xdist/blob/master/src/xdist/plugin.py#L8) gives 12:

```
from os import sched_getaffinity
len(sched_getaffinity(0))
Out[24]: 12
```

The optimal number of parallel processes though is 6. Tests run slower when using 12 processes.
Did you consider using `psutil.cpu_count(logical=False)`?

### PR DESCRIPTION
**Pull request by utapyngo**
Fixes #553.

### CODE REVIEW

File: src/xdist/plugin.py [hunk 1]
```diff
@@ -1,28 +1,13 @@
...
-            from multiprocessing import cpu_count
-    else:
-
-        def cpu_count():
-            return len(sched_getaffinity(0))
-
-    try:
-        n = cpu_count()
+        n = psutil.cpu_count(logical=False)
     except NotImplementedError:
```
Comment (graingert) [inline src/xdist/plugin.py:11]: is this something that psutil will raise?

> Return the number of logical CPUs in the system (same as os.cpu_count in Python 3.4) or None if undetermined. logical cores means the number of physical cores multiplied by the number of threads that can run on each core (this is known as Hyper Threading). If logical is False return the number of physical cores only (Hyper Thread CPUs are excluded) or None if undetermined. On OpenBSD and NetBSD psutil.cpu_count(logical=False) always return None. Example on a system having 2 physical hyper-thread CPU cores:

if it never raises, we can use `return psutil.cpu_count(logical=False) or 1`

Comment (graingert) [reply src/xdist/plugin.py:11]: I'm also concerned about:
> On OpenBSD and NetBSD psutil.cpu_count(logical=False) always return None.

should we fallback to the old code if we get a `None`?

Comment (nicoddemus) [reply src/xdist/plugin.py:11]: > should we fallback to the old code if we get a None?

Not sure, we don't officially support OpenBSD and NetBSD... your suggestion above of using `return psutil.cpu_count(logical=False) or 1` would at least prevent a crash, and I would rather keep the code cleaner. But if others disagree then I think your suggestion is on point.

Comment (graingert) [reply src/xdist/plugin.py:11]: Does this obviate the Travis workaround?

Comment (nicoddemus) [reply src/xdist/plugin.py:11]: Good question. 👍

Comment (utapyngo) [reply src/xdist/plugin.py]: image
I tested this locally on pypy35.

**Pull request merged**